src/packet.cpp

```diff
@@ -75,14 +76,17 @@ Packet::Packet(const std::array<uint8_t, 4096>& buffer, size_t bytes, const std:
 Packet::Packet(uint8_t type, uint8_t id, const std::array<uint8_t, 16>& auth, const std::vector<Attribute*>& attributes, const std::vector<VendorSpecific>& vendorSpecific)
     : m_type(type),
       m_id(id),
+      m_recalcAuth(true),
       m_auth(auth),
       m_attributes(attributes),
       m_vendorSpecific(vendorSpecific)
 {
 }
 
 Packet::Packet(const Packet& other)
-    : m_vendorSpecific(other.m_vendorSpecific)
+    : m_recalcAuth(true),
```

Comment:
Why `m_recalcAuth` is `true`? As I said earlier, it should be the same as in the `other`.